shipstation_integration/shipstation_integration/doctype/shipstation_settings/shipstation_settings.js

```diff
@@ -2,96 +2,132 @@
 // For license information, please see license.txt
 
 function company_query(frm, cdt, cdn) {
-	const row = frm.selected_doc || locals[cdt][cdn];
-	return {
-		filters: {
-			"company": row.company,
-			"is_group": 0
-		}
-	};
+    const row = frm.selected_doc || locals[cdt][cdn];
+    return {
+        filters: {
+            "company": row.company,
+            "is_group": 0
+        }
+    };
 }
 
 frappe.ui.form.on("Shipstation Settings", {
-	setup: frm => {
-		frm.set_query("shipstation_warehouses", { "shipstation_warehouse_id": ["!=", ""] });
-		frm.set_query("cost_center", "shipstation_stores", company_query);
-		frm.set_query("warehouse", "shipstation_stores", company_query);
-		frm.set_query("tax_account", "shipstation_stores", company_query);
-		frm.set_query("sales_account", "shipstation_stores", company_query);
-		frm.set_query("expense_account", "shipstation_stores", company_query);
-		frm.set_query("shipping_income_account", "shipstation_stores", company_query);
-		frm.set_query("shipping_expense_account", "shipstation_stores", company_query);
-	},
+    setup: frm => {
+        frm.set_query("shipstation_warehouses", { "shipstation_warehouse_id": ["!=", ""] });
+        frm.set_query("cost_center", "shipstation_stores", company_query);
+        frm.set_query("warehouse", "shipstation_stores", company_query);
+        frm.set_query("tax_account", "shipstation_stores", company_query);
+        frm.set_query("sales_account", "shipstation_stores", company_query);
+        frm.set_query("expense_account", "shipstation_stores", company_query);
+        frm.set_query("shipping_income_account", "shipstation_stores", company_query);
+        frm.set_query("shipping_expense_account", "shipstation_stores", company_query);
+        frm.set_query("sales_order_item_field", "options_import", function() {
+            return {
+                query: "shipstation_integration.shipstation_integration.doctype.shipstation_settings.shipstation_settings.get_item_fields"
+            };
+        });
+        frappe.meta.get_docfield("Shipstation Options Import", "sales_order_item_field", frm.docname).ignore_link_validation = true;
```

Comment:
Can you add a comment here for why we're doing this?

Reply:
you have a keen eye @Alchez. both of these code comments are related to the same issue.

I consider the use of a link control for the sales_order_item_field to be a really slick solution, but allowing both DocFields and Custom Field to be selected interchangeably requires a custom query, which throws Frappe into a tizzy when trying to validate the links, even with a "Dynamic Link" populated control.

This was my workaround. If there's a better way to handle this, I'd be glad to know how.

Reply:
@batonac, I see what you're trying to do. I think we could get away with replacing the Link control with a Select control, which would still let you populate a list to choose from as well as avoid any link validation issues.

The only downside might be the UX, since you would only be able to show the fieldnames in the list, and not the label (unless you add some special processing to separate the two, but that might be too much)